File: javascript/leetcode/is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    if (s.length == 0) {return true}
    for (var i = 0; i < t.length; i++) {
        if (t[i] == s[0]) {
            s = s.slice(1)
        }
        if (s.length == 0) {return true}
    }
    return false
};

File: python/leetcode/is-subsequence.py
class Solution:
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) == 0:
            return True
        for i in range(len(t)):
            if t[i] == s[0]:
                s = s[1:]
            if len(s) == 0:
                return True
        return False
